[samsungtv] Exception thrown when changing input source on legacy TV

Logs when attempting to change input source on a Samsung UE55D8005/XXE (`SamsungTV_SourceName` is linked to channel `sourceName`):

```
2024-05-21 22:03:54.450 [INFO ] [openhab.event.ItemCommandEvent      ] - Item 'SamsungTV_SourceName' received command HDMI1/DVI
2024-05-21 22:03:54.451 [INFO ] [penhab.event.ItemStatePredictedEvent] - Item 'SamsungTV_SourceName' predicted to become HDMI1/DVI
2024-05-21 22:03:54.452 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'SamsungTV_SourceName' changed from HDMI-CEC to HDMI1/DVI
2024-05-21 22:03:54.454 [DEBUG] [tv.internal.handler.SamsungTvHandler] - 192.168.0.41: Received channel: samsungtv:tv:office:sourceName, command: HDMI1/DVI
2024-05-21 22:03:54.454 [TRACE] [tv.internal.handler.SamsungTvHandler] - 192.168.0.41: Received: sourceName, command: HDMI1/DVI
2024-05-21 22:03:54.455 [TRACE] [rnal.service.RemoteControllerService] - 192.168.0.41: getSupportedChannelNames: [keyCode, power, channel]
2024-05-21 22:03:54.455 [TRACE] [internal.service.MainTVServerService] - 192.168.0.41: getSupportedChannelNames: [sourceName, sourceId, url, stopBrowser]
2024-05-21 22:03:54.455 [TRACE] [internal.service.MainTVServerService] - 192.168.0.41: Received channel: sourceName, command: HDMI1/DVI
2024-05-21 22:03:54.496 [TRACE] [internal.service.MainTVServerService] - 192.168.0.41: Received channel: channel, command: REFRESH
2024-05-21 22:03:54.514 [ERROR] [nal.common.AbstractInvocationHandler] - An error occurred while calling method 'ThingHandler.handleCommand()' on 'org.openhab.binding.samsungtv.internal.handler.SamsungTvHandler@bc65b9a': Duplicate key HDMI-CEC (attempted merging values 0 and 1)
java.lang.IllegalStateException: Duplicate key HDMI-CEC (attempted merging values 0 and 1)
	at java.util.stream.Collectors.duplicateKeyException(Collectors.java:135) ~[?:?]
	at java.util.stream.Collectors.lambda$uniqKeysMapAccumulator$1(Collectors.java:182) ~[?:?]
	at java.util.stream.ReduceOps$3ReducingSink.accept(ReduceOps.java:169) ~[?:?]
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197) ~[?:?]
	at java.util.stream.IntPipeline$1$1.accept(IntPipeline.java:180) ~[?:?]
	at java.util.stream.Streams$RangeIntSpliterator.forEachRemaining(Streams.java:104) ~[?:?]
	at java.util.Spliterator$OfInt.forEachRemaining(Spliterator.java:711) ~[?:?]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?]
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?]
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682) ~[?:?]
	at org.openhab.binding.samsungtv.internal.service.MainTVServerService.lambda$13(MainTVServerService.java:347) ~[?:?]
	at java.util.Optional.map(Optional.java:260) ~[?:?]
	at org.openhab.binding.samsungtv.internal.service.MainTVServerService.getSourceMap(MainTVServerService.java:346) ~[?:?]
	at org.openhab.binding.samsungtv.internal.service.MainTVServerService.setSourceName(MainTVServerService.java:306) ~[?:?]
	at org.openhab.binding.samsungtv.internal.service.MainTVServerService.handleCommand(MainTVServerService.java:173) ~[?:?]
	at org.openhab.binding.samsungtv.internal.handler.SamsungTvHandler.handleCommand(SamsungTvHandler.java:373) ~[?:?]
	at org.openhab.binding.samsungtv.internal.handler.SamsungTvHandler.handleCommand(SamsungTvHandler.java:363) ~[?:?]
	at jdk.internal.reflect.GeneratedMethodAccessor800.invoke(Unknown Source) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
	at org.openhab.core.internal.common.AbstractInvocationHandler.invokeDirect(AbstractInvocationHandler.java:147) [bundleFile:?]
	at org.openhab.core.internal.common.InvocationHandlerSync.invoke(InvocationHandlerSync.java:59) [bundleFile:?]
	at jdk.proxy187004.$Proxy187159.handleCommand(Unknown Source) [?:?]
	at org.openhab.core.thing.internal.profiles.ProfileCallbackImpl.handleCommand(ProfileCallbackImpl.java:95) [bundleFile:?]
	at org.openhab.core.thing.internal.profiles.SystemDefaultProfile.onCommandFromItem(SystemDefaultProfile.java:49) [bundleFile:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
	at org.openhab.core.internal.common.AbstractInvocationHandler.invokeDirect(AbstractInvocationHandler.java:147) [bundleFile:?]
	at org.openhab.core.internal.common.Invocation.call(Invocation.java:52) [bundleFile:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) [?:?]
	at java.lang.Thread.run(Thread.java:840) [?:?]
```

## Expected Behavior
Input source should be changed according to command.

## Current Behavior
Exception is thrown.

## Possible Solution

## Steps to Reproduce
I have tried a few times where it worked, so the current state seems to have some impact, or it might be timing-related. However, it was not very hard to reproduce, at least not right after turning on the TV.

## Context
I'm testing the new version of the binding. I have backported to changes to 4.1.x and testing on my production system.

## Your Environment
* Version used: 4.2 as of today (compiled the binding myself after #11895 was merged)

--- a/bundles/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/service/MainTVServerService.java
+++ b/bundles/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/service/MainTVServerService.java
@@ -342,10 +342,11 @@ public class MainTVServerService implements UpnpIOParticipant, SamsungTvService 
         // NodeList doesn't have a stream, so do this
         sources = Optional.of(updateResourceState("GetSourceList")).filter(a -> "OK".equals(a.get("Result")))
                 .map(a -> a.get("SourceList")).flatMap(xml -> Utils.loadXMLFromString(xml, host))
-                .map(a -> a.getDocumentElement()).map(a -> a.getElementsByTagName("Source"))
+                .map(a -> a.getDocumentElement()).map(
+                        a -> a.getElementsByTagName("Source"))
                 .map(nList -> IntStream.range(0, nList.getLength()).boxed().map(i -> (Element) nList.item(i))
                         .collect(Collectors.toMap(a -> getFirstNodeValue(a, "SourceType", ""),
-                                a -> getFirstNodeValue(a, "ID", ""))))
+                                a -> getFirstNodeValue(a, "ID", ""), (key1, key2) -> key2)))
                 .orElse(Map.of());
     }
 

